### ISSUE TITLE
Unused awaitable with TaskGroup

### ISSUE THREAD

**Issue by Dreamsorcerer** [python/mypy#15036]
```
import asyncio

async def main() -> None:
    async with asyncio.TaskGroup() as tg:
        tg.create_task(asyncio.sleep(1))
```

With unused-awaitable enabled, this code generates an unused awaitable error. TaskGroup is expected to wait on any created tasks, so there is no need to `await` on them.

Perhaps `TaskGroup` can be special-cased to ignore this error, or maybe special-case `Task` so the rule only applies when it is returned from `asyncio.create_task()`?

**Comment by alexpovel** [comment]
Just upgraded to mypy 1.5 and Python 3.11.4, and can reproduce this behaviour.

**Comment by hauntsaninja** [comment]
PR welcome!

**Comment by A5rocks** [comment]
One other case: anyio's new TaskHandles use `tg.start_soon` and are awaitable, but the outer scope already guarantees they are awaited -- no need for a lint.

In an ideal world we would only check coroutines (which definitely have to get awaited) but unfortunately people love writing `Awaitable` instead of `Coroutine`... So I guess a compromise hard-coded list of alright calls has to be done. (or we could split the error code? enabled-by-default check for unawaited coroutines (always bad, I think!) and disabled-by-default check for unawaited awaitables) I'd be happy to make a PR with whichever seems like a better approach.

**Comment by gschaffner** [comment]
> or we could split the error code? enabled-by-default check for unawaited coroutines (always bad, I think!) and disabled-by-default check for unawaited awaitables

I think it's already split into `unused-coroutine` ([playground](https://mypy-play.net/?mypy=2.3.0&python=3.14&gist=e473c7698abbc46d7b321396f58e44c5)) and `unused-awaitable` ([playground](https://mypy-play.net/?mypy=2.3.0&python=3.14&enable-error-code=unused-awaitable&gist=0a4bb3b3ce6428b1bc92a77137489e08)) like this, right?

**Comment by A5rocks** [comment]
Yeah, and `unused-coroutine` is enabled by default. I didn't realize! I guess the only thing we can do is hardcode a list of exceptions...
